pyrobosim/pyrobosim/core/dynamics.py

```diff
@@ -0,0 +1,137 @@
+        :type robot: :class:`pyrobosim.core.robot.Robot`
+        :param init_pose: The initial pose of the robot.
+        :type init_pose: :class:`pyrobosim.utils.pose.Pose`
+        :param init_vel: The initial velocity of the robot, as a [vx, vy, vtheta] array.
+        :type init_vel: :class:`numpy.array`
+        :param max_linear_velocity: The maximum linear velocity magnitude, in m/s.
+        :type max_linear_velocity: float
+        :param max_angular_velocity: The maximum angular velocity magnitude, in rad/s.
+        :type max_angular_velocity: float
+        :param max_linear_acceleration: The maximum linear acceleration magnitude, in m/s^2.
+        :type max_linear_acceleration: float
+        :param max_angular_acceleration: The maximum angular acceleration magnitude, in rad/s^2.
+        :type max_linear_acceleration: float
+        """
+        self.robot = robot
+
+        # Initial state
+        self.pose = init_pose
+        self.velocity = np.array(init_vel)
+        self.collision = False
+
+        # Velocity and acceleration limits
+        self.vel_limits = np.array(
+            [max_linear_velocity, max_linear_velocity, max_angular_velocity]
+        )
+        self.accel_limits = np.array(
+            [max_linear_acceleration, max_linear_acceleration, max_angular_acceleration]
+        )
+
+    def step(self, cmd_vel, dt, world=None, check_collisions=False):
+        """
+        Perform a single dynamics step.
+
+        :param cmd_vel: Velocity command array, in the form [vx, vy, vtheta].
+        :type cmd_vel: :class:`numpy.array`
+        :param dt: Time step, in seconds.
+        :type dt: float
+        """
+        # Trivial case of zero or None command velocities.
+        if np.count_nonzero(cmd_vel) == 0 or cmd_vel is None:
+            return
+
+        self.velocity = self.saturate_velocity_command(cmd_vel, dt)
+
+        # Dynamics
+        roll, pitch, yaw = self.pose.eul
+        sin_yaw = np.sin(yaw)
+        cos_yaw = np.cos(yaw)
+
+        vx = self.velocity[0] * cos_yaw - self.velocity[1] * sin_yaw
+        vy = self.velocity[0] * sin_yaw + self.velocity[1] * cos_yaw
+
+        target_pose = copy.copy(self.pose)
+        target_pose.x += vx * dt
+        target_pose.y += vy * dt
+        target_pose.set_euler_angles(roll, pitch, yaw + self.velocity[2] * dt)
+
+        # Check collisions
+        if check_collisions:
+            if world is None:
+                warnings.warn("Cannot check collisions without a world.")
+                return
+
+            if world.collides_with_robots(
+                target_pose, robot=self.robot
+            ) or world.check_occupancy(target_pose):
+                self.velocity = np.array([0.0, 0.0, 0.0])
+                self.collision = True
+                return
+
+        # If we made it, we succeeded
+        self.collision = False
+        self.pose = target_pose
+
+    def saturate_velocity_command(self, cmd_vel, dt):
```

Comment:
`def saturate_velocity_command(self, cmd_vel, dt):`

`saturate` feels a bit off here, since this is clipping how about something like `enforce_dynamics_limit()` borrowing from MoveIt style.

Reply:
This works for me!